### PR TITLE
Fix: context propagation in nested spans

### PR DESCRIPTION
**Pull request by GSVarsha**
Signed-off-by: Varsha GS <varsha.gs@ibm.com>

Fix: context propagation in nested spans

### CODE REVIEW

File: src/instana/tracer.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def start_span(
         record_exception: bool = True,
         set_status_on_exception: bool = True,
     ) -> InstanaSpan:
-        parent_context = span_context
+        parent_context = get_current_span().get_span_context() if span_context is None else span_context
```
Comment (pvital) [inline src/instana/tracer.py:121]: I think it's cleaner and easier to read this way. What do you think @GSVarsha ?

```suggestion
        parent_context = span_context if span_context else get_current_span().get_span_context()
```

**Review by pvital — CHANGES_REQUESTED** [review]
Nice catch @GSVarsha. Just a simple request.

File: src/instana/tracer.py (hunk 1 above)
Comment (GSVarsha) [reply src/instana/tracer.py]: Done!

**Review by pvital — APPROVED** [review]
It looks good to me.

**Review by CagriYonca — APPROVED** [review]
LGTM

**Pull request merged**